Lesson_07/homework_07.py

```diff
@@ -0,0 +1,245 @@
+"""Homework 07."""
+
+
+# task 1
+# Задача - надрукувати табличку множення на задане число, але
+# лише до максимального значення для добутку - 25.
+# Код майже готовий, треба знайти помилки та випраавити\доповнити.
+def multiplication_table(number):
+    """Multiply number.
+
+    Args:
+        number (int): Integer number
+    """
+    # Initialize the appropriate variable
+    multiplier = 1
+    # Complete the while loop condition.
+    while multiplier <= number:
+        result = number * multiplier
+        # десь тут помила, а може не одна
+        if result > 25:
+            # Enter the action to take if the result is greater than 25
+            break
+        print(str(number) + 'x' + str(multiplier) + '=' + str(result))
+
+        # Increment the appropriate variable
+        multiplier += 1
+
+
+multiplication_table(3)
+# Should print:
+# 3x1=3
+# 3x2=6
+# 3x3=9
+# 3x4=12
+# 3x5=15
+
+
+# task 2
+# Написати функцію, яка обчислює суму двох чисел.
+def sum_numbers(number1, number2):
+    """Two numbers sum.
+
+    Args:
+        number1 (int): First number
+        number2 (int): Second number
+
+    Returns:
+        int: sum of two numbers
+    """
+    return number1 + number2
+
+
+sum_result = sum_numbers(4, 8)
+
+
+# task 3
+# Написати функцію, яка розрахує середнє арифметичне списку чисел.
+def average_value(numbers_list):
+    """Average value of numbers in list.
+
+    Args:
+        numbers_list (list): List of numbers
+
+    Returns:
+        float: List of numbers average value
+    """
+    return sum(numbers_list) / len(numbers_list)
+
+
+numbers = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+average_result = average_value(numbers)
+
+
+# task 4
+# Написати функцію, яка приймає рядок та повертає його у зворотному порядку.
+def reversed_string(test_string):
+    """Reverse somme string.
+
+    Args:
+        test_string (string): Some text
+
+    Returns:
+        string: Reversed string
+    """
+    return test_string[::-1]
+
+
+txt = 'Hello World!'
+reversed_txt = reversed_string(txt)
+
+
+# task 5
+# Написати функцію, яка приймає список слів
+# та повертає найдовше слово у списку.
+def max_word(word_list):
+    """Find word with maximum length from the list.
+
+    Args:
+        word_list (list): List of words
+
+    Returns:
+        string: Word with maximum length
+    """
+    return max(str(word_list), key=len)
+
+
+words = ['automobile', 'hi', 'pencil', 'Hello', 1]
+word_result = max_word(words)
+
+
+# task 6
+# Написати функцію, яка приймає два рядки та повертає індекс першого входження
+# другого рядка у перший рядок, якщо другий рядок є підрядком першого рядка,
+# та -1, якщо другий рядок не є підрядком першого рядка.
+def find_substring(str1, str2):
+    """First entry of the second line into the first line.
+
+    Args:
+        str1 (string): First string
+        str2 (string): Second string
+
+    Returns:
+        int: Index of the first entry
+    """
+    return str1.find(str2) if str2 in str1 else -1
```

Comment:
```suggestion
    return str1.find(str2)
```
find returns -1 if nothing found -> see help(str.find)